packages/luthor/lib/src/validations/strings/string_regex_validation.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:luthor/src/validation.dart';
+
+class StringRegexValidation extends Validation {
+  final String pattern;
+  String? customMessage;
+
+  StringRegexValidation(
+    this.pattern, {
+    String? message,
+  }) : customMessage = message;
+
+  @override
+  bool call(String? fieldName, dynamic value) {
+    this.fieldName = fieldName;
+
+    if (value == null) return true;
```

Comment:
Do add a check for strings here. For my effort on having better errors, we can no longer guarantee that the value is a `String` over here since it will check all validations, even if previous ones are already false so that we can return a list of strings instead of a single string. See for example:

https://github.com/exaby73/luthor/blob/c13188ac380ee7c8b1b22a0629a6842efeaf3676/packages/luthor/lib/src/validations/strings/string_email_validation.dart#L13-L20